src/docdb/tree/DocDBStoredProceduresTreeItem.ts

```diff
@@ -77,23 +75,21 @@ export class DocDBStoredProceduresTreeItem extends DocDBTreeItemBase<StoredProce
         return this.parent.getContainerClient(client);
     }
 
-    private async validateStoredProcedureName(name: string, currStoredProcedureNames: String[]): Promise<string | null | undefined> {
-        if (name) {
-            if (name.length < 1 || name.length > 255) {
-                return "Name has to be between 1 and 255 chars long";
-            }
-            if (/[/\\?#&]/.test(name)) {
-                return localize("illegalChars", "Name contains illegal chars: /,\\,?,#,&");
-            }
-            if (name[name.length - 1] === " ") {
-                return localize("endsWithSpace", "Name cannot end with a space.");
-            }
-            if (currStoredProcedureNames.includes(name)) {
-                return localize('NameExists', 'Stored Procedure "{0}" already exists.', name);
-            }
-        } else {
-            return localize("noEmptyValue", "Name cannot be empty");
+    private validateStoredProcedureName(name: string, currStoredProcedureNames: String[]): string | undefined {
+        if (!name || name.length < 1 || name.length > 255) {
```

Comment:
You don't have to check `!name` here since it's type `string` and not something like `string | undefined`